tapioca/tapioca.py

```diff
@@ -170,14 +170,24 @@ def _get_iterator_next_request_kwargs(self):
         return self._api.get_iterator_next_request_kwargs(
             self._request_kwargs, self._data, self._response)
 
+    def _has_next(self, actual, kwargs):
+        has_more_page = "max_pages" not in kwargs or kwargs["max_pages"] > actual["page"]
+        has_more_items = "max_items" not in kwargs or kwargs["max_items"] > actual["items"]
+        return  has_more_page and has_more_items
+
     def pages(self, **kwargs):
         executor = self
         iterator_list = executor._get_iterator_list()
+        actual = {"page": 0, "items": 0}
```

Comment:
No need for dictionary here, you can define two separate variable for this: current_page, current_item